kategory/src/main/kotlin/kategory/typeclasses/Composed.kt

```diff
@@ -0,0 +1,121 @@
+package kategory
+
+/**
+ * https://www.youtube.com/watch?v=wvSP5qYiz4Y
+ */
+interface ComposedType<out F, out G>
+
+@Suppress("UNCHECKED_CAST")
+fun <F, G, A> HK<F, HK<G, A>>.lift(): HK<ComposedType<F, G>, A> = this as HK<ComposedType<F, G>, A>
+
+@Suppress("UNCHECKED_CAST")
+fun <F, G, A> HK<ComposedType<F, G>, A>.lower(): HK<F, HK<G, A>> = this as HK<F, HK<G, A>>
+
+interface ComposedFoldable<in F, in G> :
+        Foldable<ComposedType<F, G>> {
+
+    fun FF(): Foldable<F>
+
+    fun GF(): Foldable<G>
+
+    override fun <A, B> foldL(fa: HK<ComposedType<F, G>, A>, b: B, f: (B, A) -> B): B =
+            FF().foldL(fa.lower(), b, { bb, aa -> GF().foldL(aa, bb, f) })
+
+    fun <A, B> foldLC(fa: HK<F, HK<G, A>>, b: B, f: (B, A) -> B): B = foldL(fa.lift(), b, f)
+
+    override fun <A, B> foldR(fa: HK<ComposedType<F, G>, A>, lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> =
+            FF().foldR(fa.lower(), lb, { laa, lbb -> GF().foldR(laa, lbb, f) })
+
+    fun <A, B> foldRC(fa: HK<F, HK<G, A>>, lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> = foldR(fa.lift(), lb, f)
+
+    companion object {
+        operator fun <F, G> invoke(FF: Foldable<F>, GF: Foldable<G>): ComposedFoldable<F, G> =
```

Comment:
Try to see if you can inline the operator, reify F and G, and ask for foldable() for both.